ios/pcap/ipfinder.go

```diff
@@ -13,81 +16,172 @@ type NetworkInfo struct {
 	IPv6 string
 }
 
-func (n NetworkInfo) complete() bool {
-	return n.IPv6 != "" && n.Mac != "" && n.IPv4 != ""
+type PacketInfo struct {
+	Mac  string
+	IPv4 string
+	IPv6 string
+}
+
+func (n PacketInfo) complete() bool {
+	return (n.IPv6 != "" && n.Mac != "") || (n.IPv4 != "" && n.Mac != "")
 }
 
 // FindIp reads pcap packets until one is found that matches the given MAC
 // and contains an IP address. This won't work if the iOS device "automatic Wifi address" privacy
 // feature is enabled. The MAC needs to be static.
-func FindIp(device ios.DeviceEntry) (NetworkInfo, error) {
+func FindIPByMac(device ios.DeviceEntry) (NetworkInfo, error) {
 	mac, err := ios.GetWifiMac(device)
+	_ = mac
 	if err != nil {
 		return NetworkInfo{}, err
 	}
-	return findIp(device, mac)
+
+	// channel to receive PacketInfo
+	c := make(chan PacketInfo)
+	done := make(chan bool)
+
+	go func() {
+		if err := startCapture(device, c, done); err != nil {
+			log.Error(err)
+		}
+	}()
+
+	// captureDuration := time.Duration(10 * time.Second)
+	// startTime := time.Now()
+	for packet := range c {
+		if packet.Mac == mac {
+			done <- true
+			return NetworkInfo{Mac: packet.Mac, IPv4: packet.IPv4, IPv6: packet.IPv6}, nil
+		}
+	}
+
+	return NetworkInfo{}, nil
 }
 
-func findIp(device ios.DeviceEntry, mac string) (NetworkInfo, error) {
-	intf, err := ios.ConnectToService(device, "com.apple.pcapd")
+// FindIp reads pcap packets until one is found that matches the given MAC
```

Comment:
please add a comment explaining how this works

Reply:
Should be resolved in latest commit.